include/perfetto/ext/base/flat_hash_map.h

```diff
@@ -264,13 +265,19 @@ class FlatHashMap {
     if (PERFETTO_UNLIKELY(capacity_ == 0))
       return;
 
-    for (size_t i = 0; i < capacity_; ++i) {
-      const uint8_t tag = tags_[i];
-      if (tag != kFreeSlot && tag != kTombstone)
-        EraseInternal(i);
+    if (!std::is_trivially_destructible<Key>::value ||
+        !std::is_trivially_destructible<Value>::value) {
```

Comment:
I don't the is_trivially_destructible vheck buys anything. if trivially destructible, the whole for loop becomes a no-op. the compiler will figure it out.

https://godbolt.org/z/vKjMva1fj
vs
https://godbolt.org/z/168nEGvPK